storage/innobase/fut/fut0lst.cc

```diff
@@ -50,8 +49,7 @@ flst_add_to_empty(
 	ut_ad(mtr_memo_contains_page_flagged(mtr, node,
 					     MTR_MEMO_PAGE_X_FIX
 					     | MTR_MEMO_PAGE_SX_FIX));
-	len = flst_get_len(base);
-	ut_a(len == 0);
+	ut_ad(!flst_get_len(base));
```

Comment:
I think that we might want to keep this as `ut_a()` instead of `ut_ad()`, so that we can catch corruption in a non-debug server. I do not remember this assertion ever failing, but I don’t think that this is being executed too often.

Removing `len` is OK, but please keep the `ut_a()`.
Also, if you can prove that the `base` is always aligned to a word boundary, then we could add a predicate for checking if the length is 0. There is no need to adjust for the byte order, because the big-endian 0 looks exactly the same as the little-endian 0. Some non-x86 processors do not support unaligned access; that is why before adding `page_is_empty()` and `page_is_leaf()` I had to check the alignment.

Reply:
> Also, if you can prove that the base is always aligned to a word boundary
I can't: this assertion fails:
`ut_a(((uint64_t)base) % sizeof(void*) == 0);`